### PR TITLE
Add `--us-gov` flag and prod/devel US Gov environments

### PR DESCRIPTION
**Pull request by brianstrauch**
Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- Add `confluent login --us-gov` for access to the Confluent Cloud US Gov environment

Bug Fixes
- PLACEHOLDER

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
Added a new flag that logs the user in to the prod Confluent Cloud US Gov environment, as well as support for `confluent login --url devel.confluentgov-internal.com`.

Test & Review
-------------
Manual testing, waiting on account before validating further.

### CODE REVIEW

File: internal/pkg/auth/sso/auth0_utils.go [hunk 1]
```diff
@@ -4,16 +4,19 @@ import (
...
-	"devel":            "sPhOuMMVRSFFR7HfB606KLxf1RAU4SXg",
-	"fedramp-internal": "0oa7c9gkc6bHBD2OW1d7",
-	"prod":             "oX2nvSKl5jvBKVgwehZfvR4K8RhsZIEs",
-	"stag":             "8RxQmZEYtEDah4MTIIzl4hGGeFwdJS6w",
-	"test":             "00000000000000000000000000000000",
+	"cpd":          "7rG4pmRbnMn5mIsEBLAP941IE1x2rNqC",
+	"devel":        "sPhOuMMVRSFFR7HfB606KLxf1RAU4SXg",
+	"infra-us-gov": "0oa7c9gkc6bHBD2OW1d7",
+	"prod-us-gov":  "TODO",
+	"devel-us-gov": "TODO",
```
Comment (brianstrauch) [inline internal/pkg/auth/sso/auth0_utils.go:16]: @xuhui-lu, can you look these up for me?

File: internal/pkg/auth/sso/auth_state.go [hunk 2]
```diff
@@ -30,20 +30,28 @@ var (
 			ssoProviderIdentifier: "https://confluent-dev.auth0.com/api/v2/",
 			ssoProviderScope:      "email%20openid%20offline_access",
 		},
-		"fedramp-internal": {
-			ssoProviderDomain: "confluent-infra-us-gov.oktapreview.com/oauth2/v1",
+		"devel-us-gov": {
+			ssoProviderDomain: "confluent-devel-us-gov.oktapreview.com/oauth2/v1",
```
Comment (brianstrauch) [inline internal/pkg/auth/sso/auth_state.go:34]: @xuhui-lu is this correct?

File: internal/pkg/auth/sso/auth_state.go [hunk 3]
```diff
@@ -30,20 +30,28 @@ var (
...
+			ssoProviderDomain: "confluent-infra-us-gov.oktapreview.com/oauth2/v1",
+			ssoProviderScope:  "openid+profile+email+offline_access",
 		},
 		"prod": {
 			ssoProviderDomain:     "login.confluent.io/oauth",
 			ssoProviderIdentifier: "https://confluent.auth0.com/api/v2/",
 			ssoProviderScope:      "email%20openid%20offline_access",
 		},
+		"prod-us-gov": {
+			ssoProviderDomain: "confluent-prod-us-gov.oktapreview.com/oauth2/v1",
```
Comment (brianstrauch) [inline internal/pkg/auth/sso/auth_state.go:47]: @xuhui-lu is this correct? I'm not sure we're using `oktapreview` for prod.

Comment (IvanLaktyunkin) [reply internal/pkg/auth/sso/auth_state.go:47]: I believe for prod-us-gov it is confluent-prod-us-gov.okta.com, not oktapreview

Comment (brianstrauch) [reply internal/pkg/auth/sso/auth_state.go:47]:
```suggestion
			ssoProviderDomain: "confluent-prod-us-gov.okta.com/oauth2/v1",
```

File: internal/pkg/auth/sso/auth0_utils.go (hunk 1 above)
Comment (xuhui-lu) [reply internal/pkg/auth/sso/auth0_utils.go:16]: prod-us-gov: 0oa41ih4ms3TVVAT04h7
We have not provisioned the devel-us-gov tenant, so I think let's leave it as empty for now.

**Review by xuhui-lu — APPROVED** [review]
leave some nits. LGTM.

**Comment by brianstrauch** [comment]
> Were you able to get an account to do further testing?

Yes, should've updated the description!

**Pull request merged**